REST API fails when embeddings returned flag is enabled in document store

Good morning Haystack devs. Not sure if this is really a bug or I'm doing something wrong, so feel free to change the type if this issue to something else if it's not a bug.

**Describe the bug**
When the document store is forced to return the embeddings the REST API request fails because the type of the embedding is a Numpy array (`dtype=float32`) but the Pydantic dataclass of `DocumentSerialized` expects a `List[float]` and I think there is no automatic conversion between both types.

**Error message**
```
[2021-11-26 08:48:59 +0000] [10] [ERROR] Exception in ASGI application
haystack-api_1  | Traceback (most recent call last):
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 375, in run_asgi
haystack-api_1  |     result = await app(self.scope, self.receive, self.send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
haystack-api_1  |     return await self.app(scope, receive, send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/fastapi/applications.py", line 208, in __call__
haystack-api_1  |     await super().__call__(scope, receive, send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/applications.py", line 112, in __call__
haystack-api_1  |     await self.middleware_stack(scope, receive, send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/middleware/errors.py", line 181, in __call__
haystack-api_1  |     raise exc
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/middleware/errors.py", line 159, in __call__
haystack-api_1  |     await self.app(scope, receive, _send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/middleware/cors.py", line 84, in __call__
haystack-api_1  |     await self.app(scope, receive, send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/exceptions.py", line 82, in __call__
haystack-api_1  |     raise exc
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/exceptions.py", line 71, in __call__
haystack-api_1  |     await self.app(scope, receive, sender)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/routing.py", line 656, in __call__
haystack-api_1  |     await route.handle(scope, receive, send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/routing.py", line 259, in handle
haystack-api_1  |     await self.app(scope, receive, send)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/starlette/routing.py", line 61, in app
haystack-api_1  |     response = await func(request)
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/fastapi/routing.py", line 243, in app
haystack-api_1  |     is_coroutine=is_coroutine,
haystack-api_1  |   File "/usr/local/lib/python3.7/dist-packages/fastapi/routing.py", line 137, in serialize_response
haystack-api_1  |     raise ValidationError(errors, field.type_)
haystack-api_1  | pydantic.error_wrappers.ValidationError: 5 validation errors for QueryResponse
haystack-api_1  | response -> documents -> 0 -> embedding
haystack-api_1  |   value is not a valid list (type=type_error.list)
haystack-api_1  | response -> documents -> 1 -> embedding
haystack-api_1  |   value is not a valid list (type=type_error.list)
haystack-api_1  | response -> documents -> 2 -> embedding
haystack-api_1  |   value is not a valid list (type=type_error.list)
haystack-api_1  | response -> documents -> 3 -> embedding
haystack-api_1  |   value is not a valid list (type=type_error.list)
haystack-api_1  | response -> documents -> 4 -> embedding
haystack-api_1  |   value is not a valid list (type=type_error.list)
```


**To Reproduce**
Just start the REST API with `return_embedding: True` set in the document store

**System:**
 - Haystack version (commit or version number): 1.0RC1 (latest version from Github)
 - DocumentStore: Milvus
 - Reader: Farm
 - Retriever: DPR

--- a/rest_api/controller/search.py
+++ b/rest_api/controller/search.py
@@ -75,8 +75,8 @@ def _process_request(pipeline, request) -> QueryResponse:
     result = pipeline.run(query=request.query, params=params,debug=request.debug)
     
     # if any of the documents contains an embedding as an ndarray the latter needs to be converted to list of float
-    if any(isinstance(document.embedding, ndarray) for document in result['documents']):
-        for document in result['documents']:
+    for document in result['documents']:
+        if isinstance(document.embedding, ndarray):
             document.embedding = document.embedding.tolist()
     
     end_time = time.time()

--- a/rest_api/controller/search.py
+++ b/rest_api/controller/search.py
@@ -75,7 +75,7 @@ def _process_request(pipeline, request) -> QueryResponse:
     result = pipeline.run(query=request.query, params=params,debug=request.debug)
     
     # if any of the documents contains an embedding as an ndarray the latter needs to be converted to list of float
-    for document in result['documents']:
+    for document in result['documents'] or []:
         if isinstance(document.embedding, ndarray):
             document.embedding = document.embedding.tolist()
     

--- a/rest_api/controller/search.py
+++ b/rest_api/controller/search.py
@@ -2,6 +2,7 @@ import logging
 import time
 import json
 from pathlib import Path
+from numpy import ndarray
 
 from fastapi import APIRouter
 
@@ -72,6 +73,12 @@ def _process_request(pipeline, request) -> QueryResponse:
             params[key]["filters"] = _format_filters(params[key]["filters"])
 
     result = pipeline.run(query=request.query, params=params,debug=request.debug)
+    
+    # if any of the documents contains an embedding as an ndarray the latter needs to be converted to list of float
+    if any(isinstance(document.embedding, ndarray) for document in result['documents']):
+        for document in result['documents']:
+            document.embedding = document.embedding.tolist()
+    
     end_time = time.time()
     logger.info(json.dumps({"request": request, "response": result, "time": f"{(end_time - start_time):.2f}"}, default=str))
 

